### PR TITLE
siw: add `expand-all` toolbar item

### PR DESCRIPTION
**Pull request by vince-fugnitto**
#### What it does

Fixes: #7714

The pull-request introduces the `expand-all` toolbar item to the `search-in-workspace` (siw) widget to expand the result tree when it is collapsed. The toolbar toggles between the `collapse-all` and `expand-all` items based on the state of the result tree:

https://user-images.githubusercontent.com/40359487/125858322-6d6ab195-ba8b-42d9-9c74-8ee1fd7a56b8.mov

The `search-in-workspace` toolbar items are also updated to their `codicon` counterparts.

image

#### How to test

1. start the application, open the `search-in-workspace` widget and perform a search with multiple results
2. perform `collapse-all` - confirm the toolbar toggles to `expand-all` and the tree is collapsed
3. perform `expand-all` - confirm the toolbar toggles to `collapse-all` and the tree is expanded
4. repeat the steps in a multi-root workspace

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

File: packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx [hunk 1]
```diff
@@ -235,12 +235,37 @@ export class SearchInWorkspaceResultTreeWidget extends TreeWidget {
...
+            }
             if (rootFolderNode.visible) {
                 this.expansionService.collapseNode(rootFolderNode);
             }
-        });
+        }
+    }
+
+    expandAll(): void {
+        for (const [, rootFolderNode] of this.resultTree) {
```
Comment (colin-grant-work) [inline packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx:249]: Minor: I'd probably prefer `const rootFolderNode of this.resultTree.values()` to the `[, rootFolderNode]` style.

File: packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx [hunk 2]
```diff
@@ -235,12 +235,37 @@ export class SearchInWorkspaceResultTreeWidget extends TreeWidget {
...
+    }
+
+    areResultsCollapsed(): boolean {
+        for (const [, rootFolderNode] of this.resultTree) {
+            for (const fileNode of rootFolderNode.children) {
+                const isChildCollapsed = ExpandableTreeNode.isCollapsed(fileNode);
+                if (!isChildCollapsed) {
+                    return false;
+                }
+            }
```
Comment (colin-grant-work) [inline packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx:266]: Could do:
```typescript
if (rootFolderNode.some(fileNode => !ExpandableTreeNode.isCollapsed(fileNode))) {
    return false;
}
```
but no real preference here :-).

**Review by colin-grant-work — APPROVED** [review]
Works well for me, and the code looks good. Two very optional comments.

**Comment by colin-grant-work** [comment]
> Manually collapsing all nodes should change the icon to `Expand all` but it doesn't,
> Not a major problem but it would be good to investigate it.

Good catch. To make that work, we'll need to forward the expansion events from the tree to the `onDidChange` for the tabbar items.

**Comment by vince-fugnitto** [comment]
@alvsan09 I believe it is addressed if you'd like to retry :+1:

**Review by alvsan09 — APPROVED** [review]
It works as expected and the code looks good to me ! :+1: 

Symbolic approval as I am not a committer..

**Comment by colin-grant-work** [comment]
I reaffirm my earlier approval in light of the new changes. 👍

**Pull request merged**